modules/plugin/jdbc/jdbc-postgis/src/main/java/org/geotools/data/postgis/PostgisNGDataStoreFactory.java

```diff
@@ -302,10 +312,11 @@ protected String getJDBCUrl(Map<String, ?> params) throws IOException {
         String host = (String) HOST.lookUp(params);
         String db = (String) DATABASE.lookUp(params);
         int port = (Integer) PORT.lookUp(params);
-        String url = "jdbc:postgresql" + "://" + host + ":" + port + "/" + db;
+        Boolean reWriteBatchedInserts = (Boolean) REWRITE_BATCHED_INSERTS.lookUp(params);
+        String url = "jdbc:postgresql" + "://" + host + ":" + port + "/" + db + "?reWriteBatchedInserts=" + reWriteBatchedInserts;
         SslMode mode = (SslMode) SSL_MODE.lookUp(params);
         if (mode != null) {
-            url = url + "?sslmode=" + mode + "&binaryTransferEnable=bytea";
+            url = url + "&sslmode=" + mode.value + "&binaryTransferEnable=bytea";
```

Comment:
This should be a separate JIRA ticket and PR with its own unit test.

Reply:
updated the testGetJDBCUrl to exclude the sslmode param for now